src/foam/nanos/ruler/RuleEngine.java

```diff
@@ -84,6 +84,7 @@ public void execute(List<Rule> rules, FObject obj, FObject oldObj) {
       compoundAgency.execute(x_);
     } catch (Exception e) {
       logger.error(e.getMessage(), e);
+      throw e;
```

Comment:
this is occuring asynchronously, there is nothing to catch the exception. Reporting is all we can do.  Perhaps we can add it to the rule - so a report of what failed. but we can't throw from here.